plugins/analysis/client/plugin.go

```diff
@@ -8,24 +8,41 @@ import (
 	"time"
 
 	"github.com/iotaledger/goshimmer/packages/shutdown"
-	"github.com/iotaledger/goshimmer/plugins/analysis/types/heartbeat"
+	"github.com/iotaledger/goshimmer/plugins/analysis/packet"
 	"github.com/iotaledger/goshimmer/plugins/autopeering"
 	"github.com/iotaledger/goshimmer/plugins/autopeering/local"
 	"github.com/iotaledger/goshimmer/plugins/config"
 	"github.com/iotaledger/hive.go/daemon"
 	"github.com/iotaledger/hive.go/logger"
 	"github.com/iotaledger/hive.go/network"
 	"github.com/iotaledger/hive.go/node"
+	flag "github.com/spf13/pflag"
 )
 
-var log *logger.Logger
-var connLock sync.Mutex
+const (
+	// PluginName is the name of  the analysis client plugin.
+	PluginName = "Analysis-Client"
+	// CfgServerAddress defines the config flag of the analysis server address.
+	CfgServerAddress = "analysis.client.serverAddress"
+	// defines the report interval of the reporting in seconds.
+	reportIntervalSec = 5
+)
+
+func init() {
+	flag.String(CfgServerAddress, "ressims.iota.cafe:188", "tcp server for collecting analysis information")
+}
+
+var (
+	// Plugin is the plugin instance of the analysis client plugin.
+	Plugin   = node.NewPlugin(PluginName, node.Enabled, run)
+	log      *logger.Logger
+	connLock sync.Mutex
+)
 
-// Run runs the plugin.
-func Run(plugin *node.Plugin) {
-	log = logger.NewLogger("Analysis-Client")
-	daemon.BackgroundWorker("Analysis Client", func(shutdownSignal <-chan struct{}) {
-		ticker := time.NewTicker(ReportInterval * time.Second)
+func run(_ *node.Plugin) {
+	log = logger.NewLogger(PluginName)
+	_ = daemon.BackgroundWorker(PluginName, func(shutdownSignal <-chan struct{}) {
```

Comment:
Instead of ignoring potential errors from `BackgroundWorker` consider logging them

Reply:
Made it panic.